### ISSUE TITLE
Add Possibility to Auto Fix Tearing

### ISSUE THREAD

**Issue by liberodark** [solus-project/linux-driver-management#12]
Hello,

On Intel chipset have tearing and on nvidia.
But have a simple possibility to fix that in auto.
Is possible for view to add fonctionality on solus 4 for fix tearing ?

Cordialy

**Comment by ikeydoherty** [comment]
Certainly possible, we control GLX after all. Depends on exactly how we we want to expose this, though.
What is the workflow, for example? How is this activated? etc.

**Comment by andrebrait** [comment]
Are you talking about PRIME Sync? As in having smooth graphics on Intel IGP but tearing when using the NVIDIA GPU? 

If you're having tearing even on Intel, it.might be related to not having a compositor running.

**Comment by ikeydoherty** [comment]
For PRIME mostly its a case of being SoL right now.

**Comment by andrebrait** [comment]
Yeah, they still have tons of work ahead of them to get PRIME Sync to work well. They're currently dealing with instabilities and stuff like two cursors being present in the screen when it's activated. I've had crashes in some games and whatnot with it turned on.

I wanted to know if OP had problems with that or if it was with another setup because other than PRIME Sync, tearing should  already be fixed by now, if using a compositor.

**Comment by liberodark** [comment]
For intel is more easy IGP from intel all linux dristribution have tearing on intel chipset but have solution for fix that

**Comment by liberodark** [comment]
The best is the addition of putting a tearfree or not!
But first you have to have a hardware detection.
After that it's pretty generic.
But since it is a very common issue under linux, it would be a first linux that does not have tearing and can easily fix it without the command line typing it would be wahoo.
A saving of time and especially an accessibility to the biggest numbers.
Do not you believe?

**Comment by ikeydoherty** [comment]
Forcing DRI2 will absolutely gimp performance where DRI3 is available. Likewise force enabling UXA is a Very Bad Thing where SNA would be used, a catastrophic performance drop. The only option there of use is `Option "TearFree"` - which even then would need testing across a **wide range** of Intel devices to ensure it doesn't introduce regressions.
